modules/ting/src/Search/TingSearchRequest.php

```diff
@@ -0,0 +1,501 @@
+   *   The strategy to use when searching.
+   */
+  public function __construct($search_strategy) {
+    $this->searchStrategy = $search_strategy;
+  }
+
+  /**
+   * Sets the maximum number of results to return.
+   *
+   * @param int $max
+   *   The maximum value.
+   *
+   * @return \Ting\Search\TingSearchRequest
+   *   Current search query object.
+   */
+  public function setCount($max) {
+    $this->numResults = $max;
+    return $this;
+  }
+
+  /**
+   * Gets the current maximum number of results to return.
+   *
+   * @return int|NULL
+   *   The maximum value. NULL if the value has not been set.
+   */
+  public function getCount() {
+    return $this->numResults;
+  }
+
+  /**
+   * Performs the search-query.
+   *
+   * @return \Ting\Search\TingSearchResultInterface
+   *   The response.
+   */
+  public function execute() {
+    // Delegate to the strategy.
+    return $this->searchStrategy->executeSearch($this);
+  }
+
+  /**
+   * The part of the query that should be interpreted as a fulltext query.
+   *
+   * @return string
+   *   The query fragment.
+   */
+  public function getFullTextQuery() {
+    return $this->fullTextQuery;
+  }
+
+  /**
+   * Sets a string that should be interpreted as a fulltext query.
+   *
+   * The query may still contain more "advanced" parts such as a field filter.
+   *
+   * @param string $full_text_query
+   *   Any string.
+   *
+   * @return TingSearchRequest
+   *   the current query object.
+   */
+  public function setFullTextQuery($full_text_query) {
+    $this->fullTextQuery = $full_text_query;
+    return $this;
+  }
+
+  /**
+   * Returns whether searches should be fuzzy.
+   *
+   * @return bool
+   *   Whether the search should be fuzzy.
+   */
+  public function isFuzzy() {
+    return $this->fuzzy;
+  }
+
+  /**
+   * Sets whether searches should be fuzzy.
+   *
+   * This will in particular affect full-text queries specified via
+   * setFullTextQuery()
+   *
+   * @param bool $fuzzy
+   *   The flag.
+   *
+   * @return TingSearchRequest
+   *   the current query object.
+   */
+  public function setFuzzy($fuzzy) {
+    $this->fuzzy = $fuzzy;
+    return $this;
+  }
+
+  /**
+   * Maximum number of terms to return per facet.
+   *
+   * @return int
+   *   The maximum.
+   */
+  public function getTermsPerFacet() {
+    return $this->termsPerFacet;
+  }
+
+  /**
+   * Sets the maximum number of terms to return per facet.
+   *
+   * @param int $terms_per_facet
+   *   The maximum number of terms to return pr facet.
+   *
+   * @return TingSearchRequest
+   *   The current query object.
+   */
+  public function setTermsPrFacet($terms_per_facet) {
+    $this->termsPerFacet = $terms_per_facet;
+    return $this;
+  }
+
+  /**
+   * Get the page the search result should start at.
+   *
+   * @return int
+   *   The page-number, defaults to 1.
+   *
+   * @return TingSearchRequest
+   *   the current query object.
+   */
```

Comment:
This function documentation has two @return statement?

Reply:
:+1: